hueplusplus/include/HueDeviceTypes.hpp

```diff
@@ -0,0 +1,123 @@
+/**
+    \file DeviceTypes.h
+    Copyright Notice\n
+    Copyright (C) 2020  Jan Rogall      - developer\n
+    Copyright (C) 2020  Moritz Wirger   - developer\n
+
+    This file is part of hueplusplus.
+
+    hueplusplus is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    hueplusplus is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with hueplusplus.  If not, see <http://www.gnu.org/licenses/>.
+**/
+
+
+#ifndef _HUEDEVICETYPES_H
+#define _HUEDEVICETYPES_H
+
+#include <set>
+#include <string>
+
+#include "HueExceptionMacro.h"
+#include "HueLight.h"
+
+static const std::set<std::string> c_EXTENDEDCOLORLIGHT_GAMUTB_TYPES = {
+    "LCT001", "LCT002", "LCT007", "LLM001"
+};
+
+static const std::set<std::string> c_EXTENDEDCOLORLIGHT_GAMUTC_TYPES = {
+    "LCT010", "LCT011", "LCT012", "LCT014",
+    "LCT015", "LCT016", "LLC020", "LST002"
+};
+
+static const std::set<std::string> c_EXTENDEDCOLORLIGHT_GAMUTA_TYPES = {
+  "LST001", "LLC005" , "LLC006", "LLC007",
+    "LLC010", "LLC011", "LLC012", "LLC013",
+    "LLC014"
+};
+
+static const std::set<std::string> c_DIMMABLELIGHT_NO_COLOR_TYPES = {
+    "LWB004", "LWB006", "LWB007", "LWB010",
+    "LWB014", "LDF001", "LDF002", "LDD001",
+    "LDD002", "MWM001"
+};
+
+static const std::set<std::string> c_TEMPERATURELIGHT_TYPES = {
+    "LLM010", "LLM011", "LLM012", "LTW001", "LTW004",
+    "LTW010", "LTW011", "LTW012", "LTW013", "LTW014",
+    "LTW015", "LTP001", "LTP002", "LTP003", "LTP004",
+    "LTP005", "LTD003", "LTF001", "LTF002", "LTC001",
+    "LTC002", "LTC003", "LTC004", "LTC011", "LTC012",
+    "LTD001", "LTD002", "LFF001", "LTT001", "LDT001"
+};
+
+static const std::set<std::set<std::string>> c_ALL_DEVICE_SETS = {
+    c_EXTENDEDCOLORLIGHT_GAMUTB_TYPES,
+    c_EXTENDEDCOLORLIGHT_GAMUTC_TYPES,
+    c_EXTENDEDCOLORLIGHT_GAMUTA_TYPES,
+    c_DIMMABLELIGHT_NO_COLOR_TYPES,
+    c_TEMPERATURELIGHT_TYPES
+};
+
+struct MakeHueLight {
+
+    template<typename Commands,
+        typename BrightnessStrategy,
+        typename ColorHueStrategy,
+        typename ColorTemperatureStrategy,
+        typename SimpleColorTemperatureStrategy>
+    auto operator()(std::string type, int id, Commands commands, BrightnessStrategy simpleBrightnessStrategy,
```

Comment:
This function does not need to be a template, and can also be moved into a cpp file. That way more types can be added easier.
Also it could be a static function.